src/promise-katas-2.js

```diff
@@ -25,17 +25,48 @@ fetch("joke", "question").then(<insert your callback function>)
 // 1 Create a function that uses the fetch function to make a request to the "food" URL and returns
 // the data - expected return value "Cheese" of type String
 
-const food = () => {};
+// THIS does work but uses the .then
+/* const food = async () => {
+  return await fetch("food")
+    .then(
+      value => { return value.data; }
+    );
+}; */
+
+//THIS works but does not use the .then
+const food = async () => {
+   const { data } = await fetch("food");
+   return data;
+ };
+
 
 // 2 Create a function that uses the fetch function to make a request to the "cats" URL and returns
 // a list of cats in alphabetical order - expected return value ["Bandit", "Berry", "Puss in boots", "Smokey"] of type Array
 
-const cat = () => {};
+const cat = async () => {
+  const promise = await fetch("cats");
+  console.log(promise.data);
+
+   return promise.data.cats.sort();
+  };
+  /* const cat = async () => {
+    const { data } = await fetch("cats");
+    console.log(data);
+
+   return data.cats.sort();
+  }; */
 
 // 3 Create a function that uses the fetch function to make a request to the "dogs" URL and returns
 // the naughtiest dog - expected return value {name: "Mutley", naughty: 10} of type Object
 
-const dog = () => {};
+const dog = async () => {
+const { data } = await fetch("dogs");
+console.log(data);
+
+const naughtiestDog = data.dogs.reduce( (acc, curr) => acc.naughty > curr.naughty ? acc : curr);
+  console.log("Naughty Doggy", naughtiestDog);
+  return(naughtiestDog);
```

Comment:
As before, you should use the `.then` function after a fetch and remove all console logs. Using the same logic you have here you could re-write this to:
```
  return fetch("dogs").then((res) => {
    const naughtiestDog = res.data.dogs.reduce((acc, curr) =>
      acc.naughty > curr.naughty ? acc : curr
    );
    return naughtiestDog;
  });
```